src/shared/udev-util.c

```diff
@@ -154,6 +154,9 @@ static int device_monitor_handler(sd_device_monitor *monitor, sd_device *device,
         assert(data->sysname || data->devlink);
         assert(!data->device);
 
+        if (device_for_action(device, DEVICE_ACTION_REMOVE))
+                return 0;
+
```

Comment:
Hmm, this commit needs more justification. We *may* care about initialization mostly, but we get a REMOVE event, wouldn't that mean that the waiting for the device is done too?